forc-pkg/src/manifest/mod.rs

```diff
@@ -206,15 +208,45 @@ pub struct Network {
     pub url: String,
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+pub struct HexSalt(pub fuel_tx::Salt);
+
+impl FromStr for HexSalt {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // cut 0x from start.
+        let normalized = &s[2..];
+        if &s[0..2] != "0x" {
+            anyhow::bail!("hex salt definition needs to start with 0x.")
+        }
+        let salt: fuel_tx::Salt =
+            fuel_tx::Salt::from_str(normalized).map_err(|e| anyhow::anyhow!("{e}"))?;
+        let hex_salt = Self(salt);
+        Ok(hex_salt)
+    }
+}
+
+impl Display for HexSalt {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let salt = self.0;
+        write!(f, "0x{}", salt)
```

Comment:
```suggestion
        write!(f, "{}", salt)
```

The test is failing with this string:
`0x0x0000000000000000000000000000000000000000000000000000000000000000`